.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,41 @@
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main]
```

Comment:
Is this redundant since the `main` branches are protected and that any pushes onto `main` are necessarily done by pull requests? I might advocate for triggering on pushes to release branches such as `rXX.YY` since these are not protected.

Reply:
Thank you for reviewing this PR. That is a really good point! I've changed this line to `branches: ['r[0-9][0-9].[0-9][0-9]']`, which should cover branches like r22.01 and r23.06.

I asked a question to the GitHub team [here](https://github.com/github/branch-deploy/issues/165). In the CodeQL template they provided, it said pull request branches need to be a subset of push branches in this configuration file, but [local testing](https://github.com/github/branch-deploy/issues/165) found that to be untrue and actually duplicate the workflow if the branch matches both patterns.

Assuming that is correct, it  could be good to update both this workflow and the CodeQL workflow to run for all PRs (or at least those to main and release branches).

Is there a reason we don't protect release branches? It looks like it can be done here for pattern matching. I'll open a conversation offline about this, because it might make sense to require PRs for both. In that case, we wouldn't need to run on merge.

Reply:
Based on our offline conversation, release branches have other forms of protection. As a result, I'm going to modify the pre-commit hooks to run for all pull requests. I don't see much risk there.

Based on testing, it seems like there's no reason that pull request branches need to be a subset of push branches for this workflow automation. I'm going to change this PR to run pre-commit and codeQL for all pull requests. If we merge it and see that actions keep running as expected (or I hear back from the GitHub team confirming that there's no risk), I'll update the PRs in the other repos to do the same.